crates/pallet-subspace/src/lib.rs

```diff
@@ -659,7 +659,7 @@ impl<T: Config> Pallet<T> {
 
     /// Total number of pieces in the blockchain
     pub fn total_pieces() -> u64 {
-        u64::from(RecordsRoot::<T>::count()) * u64::from(MERKLE_NUM_LEAVES)
+        u64::from(RecordsRoot::<T>::count()).max(1) * u64::from(MERKLE_NUM_LEAVES)
```

Comment:
This doesn't feel right. I was thinking of fixing this on RPC layer instead.

Can't say I'm strongly opposed to it, but it will have to be handled on RPC layer either way since these changes only take effect after runtime upgrade.